sdk/azcore/fake/fake.go

```diff
@@ -43,507 +31,110 @@ type TokenCredential struct {
 // SetError sets the specified error to be returned from GetToken().
 // Use this to simulate an error during authentication.
 func (t *TokenCredential) SetError(err error) {
-	t.err = &nonRetriableError{err}
+	t.err = shared.NonRetriableError(err)
 }
 
 // GetToken implements the azcore.TokenCredential for the TokenCredential type.
 func (t *TokenCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
 	if t.err != nil {
-		return azcore.AccessToken{}, &nonRetriableError{t.err}
+		return azcore.AccessToken{}, shared.NonRetriableError(t.err)
 	}
 	return azcore.AccessToken{Token: "fake_token", ExpiresOn: time.Now().Add(24 * time.Hour)}, nil
 }
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 // Responder represents a scalar response.
-type Responder[T any] struct {
-	resp    T
-	content ResponseContent
-	opts    SetResponseOptions
-}
+type Responder[T any] exported.Responder[T]
 
 // SetResponse sets the specified value to be returned.
+//   - httpStatus is the HTTP status code to be returned
 //   - resp is the response to be returned
 //   - o contains optional values, pass nil to accept the defaults
-func (r *Responder[T]) SetResponse(resp T, o *SetResponseOptions) {
-	r.resp = resp
-	if o != nil {
-		r.opts = *o
-	}
-}
-
-// SetHeader sets the specified header key/value pairs to be returned.
-// Call multiple times to set multiple headers.
-func (r *Responder[T]) SetHeader(key, value string) {
-	if r.content.Header == nil {
-		r.content.Header = http.Header{}
-	}
-	r.content.Header.Set(key, value)
+func (r *Responder[T]) SetResponse(httpStatus int, resp T, o *SetResponseOptions) {
```

Comment:
I have trepidations about the `httpStatus` parameter.  Given that `SetResponse()` is _always_ for the success case, we've now opened the possibility of the customer specifying the wrong HTTP status code, either the wrong 2xx status code or some other code that doesn't make sense for the success case.

Reply:
Per discussion, we will leave this as-is.  To help push customers into the pit of success, we will codegen comments containing the list of successful HTTP status codes and return a descriptive error message when there's a mismatch.